### DISCUSSION TITLE
Want to open joyride modal with some delay

### DISCUSSION THREAD

**Discussion by gaurav0909-max** [gilbarbara/react-joyride#1087] [General]
Hey folks and @gilbarbara ,
I want your help in this here I attached two steps where I need your attention
I want to assign some delay time to open the joyride modal on the next page because It doesn't get the classname of button and display the modal on top left side as show in image
Is there any way to fix this issue?
![Screenshot (105)](https://github.com/user-attachments/assets/4f97a145-d3cc-4130-bb00-042b8c608482)
![Screenshot (106)](https://github.com/user-attachments/assets/e687147f-c99c-44bf-bc60-2cd8546520eb)

**Comment by gilbarbara** [comment]
Hey @gaurav0909-max 

I'm not entirely sure what's happening in the second image, but you'll probably need to handle the tour yourself with the `stepIndex` prop and the `callback`.
Check the multi-route [example](https://codesandbox.io/p/devbox/github/gilbarbara/react-joyride-demo?file=%2Fsrc%2FMultiRoute%2Findex.tsx) in the demo.
